### PR TITLE
add `a_sum_ratio_j()` analysis function

### PR DESCRIPTION
**Pull request by vikram-rawat**
Fixes - #484

Adds a new rtables analysis function that computes sum, unique sum, and their ratios against a denominator column. Useful for tables that need event counts and event rates per exposure (e.g., events / person-days).

### What changed

- `R/a_sum_ratio_j.R` — new exported function with 4 stats: `sum`, `sum_unique`, `ratio`, `ratio_unique`
- `R/junco_utils_default_stats_formats_labels.R` — registered stats, formats, and labels in the central registry
- `tests/testthat/test-a_sum_ratio_j.R` — 12 tests covering happy path, edge cases (empty df, all-NA, zero denominator), custom labels/formats, and input validation

### Why

There was no junco function for computing a simple sum-over-sum ratio within an rtables layout. Programmers had to write custom `afun`s for tables needing event counts divided by exposure totals. This function standardizes that pattern with built-in support for unique-row handling via `id_var`.

## Checks

- [x] (Have you updated the NEWS.md ?)
- [x] (Have you added proper tests for new functions/features ?)
- [x] (Have you added new functions to the pkgdown.yml ?)
- [x] (Have you run document() on new functions ?)

### CODE REVIEW

File: tests/testthat/test-a_sum_ratio_j.R [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+# S01 appears twice in ARM=A — this tests that sum_unique keeps unique rows correctly
+
+df <- data.frame(
+  USUBJID = c("S01", "S01", "S02", "S03"),
+  ARM = factor(c("A", "A", "A", "B")),
+  EVENTS = c(1, 1, 0, 1),
+  DAYS = c(10, 10, 20, 15)
+)
+
+# Expected results per ARM:
```
Comment (munoztd0) [inline tests/testthat/test-a_sum_ratio_j.R]: please remove expected commented results

File: tests/testthat/test-a_sum_ratio_j.R [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+#          |   A          |   B
+# ---------+--------------+---------
+# sum      |   2          |   1
+# sum_uniq |   1          |   1
+# ratio    |   2/40=0.05  |   1/15=0.067
+# rat_uniq |   1/30=0.033 |   1/15=0.067
+
+# Start of tests ----
+
+# Happy path: all 4 stats together in one layout
```
Comment (munoztd0) [inline tests/testthat/test-a_sum_ratio_j.R]: removed commented header and/or make sure its in the test message

File: R/a_sum_ratio_j.R [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    # denom_by column must also be numeric
+    checkmate::assert_numeric(df[[denom_by]], .var.name = denom_by)
+  }
+  if (needs_unique) {
+    checkmate::assert_string(id_var)
+    checkmate::assert_names(colnames(df), must.include = id_var)
+  }
+
+  # Helpers (NAs always removed) -----------------------------------------------
+  # Plain sum across all rows
```
Comment (munoztd0) [inline R/a_sum_ratio_j.R]: OVERALL COMMENT: The comment here repeats exactly what the function name (sum_unique) already tells us. Let's remove it to keep the code clean. 

In a standalone script, line-by-line comments describing the steps are fine, but in an R package, we want functions to be self-documenting. Try to avoid comments that explain what the code is doing mechanically. Instead, save internal comments for explaining the why (like business logic or complex workarounds) if the code itself can't show it. Anything else should go into the function's formal Roxygen documentation headers. Please review all the comments

**Review by munoztd0 — CHANGES_REQUESTED** [review]
overrall comments change requested

File: tests/testthat/test-a_sum_ratio_j.R (hunk 1 above)
Comment (vikram-rawat) [reply tests/testthat/test-a_sum_ratio_j.R]: thanks for the comment. fixed it.

File: tests/testthat/test-a_sum_ratio_j.R (hunk 2 above)
Comment (vikram-rawat) [reply tests/testthat/test-a_sum_ratio_j.R]: removed it.

File: R/a_sum_ratio_j.R (hunk 3 above)
Comment (vikram-rawat) [reply R/a_sum_ratio_j.R]: removed all comments from both the files.

**Comment by vikram-rawat** [comment]
removed almost all comments from both the files.

**Comment by munoztd0** [comment]
Fix CICD checks

File: tests/testthat/_snaps/docx_exporter_functions.md
Comment (munoztd0) [inline tests/testthat/_snaps/docx_exporter_functions.md:1]: you should NOT push this !!!!!!!!!!!!!!! please revert ! (those are false positives snapshot diffs due to windows) NEVER PUSH SNAPSHOT CHANGES IF YOU DONT HAVE CHANGED

File: tests/testthat/_snaps/estimate_proportion_diff.md
Comment (munoztd0) [inline tests/testthat/_snaps/estimate_proportion_diff.md:1]: what is this change .. same , please revert!!

**Review by munoztd0 — CHANGES_REQUESTED** [review]
do not update snapshots willy nelly , revert immediately

File: tests/testthat/_snaps/docx_exporter_functions.md
Comment (vikram-rawat) [reply tests/testthat/_snaps/docx_exporter_functions.md:1]: removed snapshots.

**Comment by vikram-rawat** [comment]
removed both the snapshots.
